platform/pulpcore/app/models/task.py

```diff
@@ -163,6 +163,23 @@ class Task(Model):
     parent = models.ForeignKey("Task", null=True, related_name="spawned_tasks")
     worker = models.ForeignKey("Worker", null=True, related_name="tasks")
 
+    def set_worker(self, worker):
+        """
+        Set the worker Foreign Key on this Task, save it, and log output in warning cases.
+
+        This updates the :attr:`worker`.
```

Comment:
I was assuming it was correct as other attributes on the same model are referenced the same way, but I will try to render the docs to make sure.